### PR TITLE
cleanup estimate fee

### PR DESCRIPTION
**Pull request by nitsujlangston**
* fix rounding issue on feePerKb
* add feePerByte concept (from bitcore-lib-cash)
* only add change for fee if still economical after increased size due to additional change output

### CODE REVIEW

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -80,6 +80,9 @@ Transaction.NLOCKTIME_MAX_VALUE = 4294967295;
 // Value used for fee estimation (satoshis per kilobyte)
 Transaction.FEE_PER_KB = 100000;
 
+// Value used for fee estimation (satoshis per byte)
+Transaction.FEE_PER_BYTE = 1;
```
Comment (jguthrie7) [inline lib/transaction/transaction.js]: I don't see what the point of this constant is. It is unused, and it doesn't match with the existing `FEE_PER_KB`

Comment (nitsujlangston) [reply lib/transaction/transaction.js]: agreed, removed

File: lib/transaction/transaction.js [hunk 2]
```diff
@@ -968,10 +984,19 @@ Transaction.prototype.getFee = function() {
...
   var estimatedSize = this._estimateSize();
   var available = this._getUnspentValue();
-  return Transaction._estimateFee(estimatedSize, available, this._feePerKb);
+  var feeRate = this._feePerByte || Math.ceil((this._feePerKb || Transaction.FEE_PER_KB) / 1000);
+  function getFee(size) {
+    return size * feeRate;
+  }
+  var noChangeFee = getFee(estimatedSize);
+  var changeFee = getFee(Transaction.CHANGE_OUTPUT_MAX_SIZE);
+  if (available <= noChangeFee + changeFee) {
```
Comment (nitsujlangston) [inline lib/transaction/transaction.js]: Need to check if there is already a change output

File: bitcore-lib.js [hunk 3]
```diff
@@ -10862,10 +11121,22 @@ Transaction.prototype.getFee = function() {
 /**
  * Estimates fee from serialized transaction size in bytes.
  */
-Transaction.prototype._estimateFee = function() {
+Transaction.prototype._estimateFee = function () {
   var estimatedSize = this._estimateSize();
   var available = this._getUnspentValue();
-  return Transaction._estimateFee(estimatedSize, available, this._feePerKb);
+  var feeRate = this._feePerByte || Math.ceil((this._feePerKb || Transaction.FEE_PER_KB) / 1000);
```
Comment (nitsujlangston) [inline bitcore-lib.js:11127]: Should handle decimal fee rates

**Comment by nitsujlangston** [comment]
moved to main bitcore monorepo

**Pull request closed without merging**